authors/apps/authentication/views.py

```diff
@@ -73,3 +77,70 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+class EmailSentAPIView(generics.CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = EmailSerializer
+
+    def post(self, request):
+        """
+        here, the user provides email to be used to get a link. The email must be registered,
+        token gets generated and sent to users via link.
+        """
+        email = request.data.get('email', {})
+        try:
+            if request.data['email'].strip() == "":
+                message = {"message":"email field cannot be empty"}
+                return Response(message, status=status.HTTP_400_BAD_REQUEST)
+        except KeyError:
+            message = {"message":"email field should be provided"}
+            return  Response(message, status=status.HTTP_400_BAD_REQUEST)
+        user = User.objects.filter(email=email).first()
+        if user is None:
+            message = {"message":"The Email provided is not registered"}
+            return Response(message, status=status.HTTP_400_BAD_REQUEST)
+        serializer = self.serializer_class(data={"email":email})
+        serializer.is_valid(raise_exception=True)
+        token_generator = PasswordResetTokenGenerator()
+        token = token_generator.make_token(user)
+        message = {"message":"We've sent a password reset link to your email"}
+        subject = "Password reset"
+        body = render_to_string('password_reset.html', {
+            'link': 'http://127.0.0.1:8000/api/users/password_reset?token=' + token,
```

Comment:
Let's have the base URL coming from the environment variables  through the `settings.py`